### PR TITLE
Fix capitilzation of default engine example

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Updates the capitalization of new default for `engine` setting, according to developer feedback.

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
This PR does not need to be backported since it is related to a 2.18 change, #8620

File: _search-plugins/knn/knn-index.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Mapping parameter | Required | Default | Updatable | Description
 :--- | :--- | :--- | :--- | :---
 `name` | true | n/a | false | The identifier for the nearest neighbor method.
 `space_type` | false | l2 | false | The vector space used to calculate the distance between vectors. Note: This value can also be specified at the top level of the mapping.
-`engine` | false | Faiss  | false | The approximate k-NN library to use for indexing and search. The available libraries are Faiss, nmslib, and Lucene.
+`engine` | false | faiss  | false | The approximate k-NN library to use for indexing and search. The available libraries are Faiss, nmslib, and Lucene.
```
Comment (VijayanB) [inline _search-plugins/knn/knn-index.md:95]:
```suggestion
`engine` | false | faiss  | false | The approximate k-NN library to use for indexing and search. The available libraries are faiss, nmslib, and lucene.
```

Comment (VijayanB) [reply _search-plugins/knn/knn-index.md:95]: shall we make this change as well?

Comment (Naarcha-AWS) [inline _search-plugins/knn/knn-index.md:95]: Lets switch it to code format.

```suggestion
`engine` | false | faiss  | false | The approximate k-NN library to use for indexing and search. The available libraries are `faiss`, `nmslib`, and `lucene`.
```

**Comment by VijayanB** [comment]
Can this be merged?

**Pull request merged**